nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/BatchUploadObject.java

```diff
@@ -292,10 +307,14 @@ public Object execute(@PathParam(REQUEST_BATCH_ID) String batchId, @PathParam(RE
 
     protected Object executeBatch(String batchId, String fileIdx, String operationId, HttpServletRequest request,
             ExecutionRequest xreq) throws UnsupportedEncodingException {
-        RequestContext.getActiveContext(request).addRequestCleanupHandler(req -> {
-            BatchManager bm = Framework.getService(BatchManager.class);
-            bm.clean(batchId);
-        });
+
+        if (!Boolean.parseBoolean(
+            RequestContext.getActiveContext(request).getRequest().getHeader(BatchManagerConstants.NO_DROP_FLAG))) {
+            RequestContext.getActiveContext(request).addRequestCleanupHandler(req -> {
+                BatchManager bm = Framework.getService(BatchManager.class);
+                bm.clean(batchId);
+            });
+        }
```

Comment:
Currently, it is up to the user to drop the batch if he/she chooses to explicitly pass the header to prevent the batch from being dropped. But we could come up with some sort of cron job to do some clean up. However, this would have to be carefully thought because we have no way of knowing if the user is still going to use the batch or not. We would have to come up with some sort of heuristic.